syncpack.config.cjs

```diff
@@ -0,0 +1,195 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	indent: "\t",
+
+	// Don't set dep versions based on the version of the package in the workspace
+	workspace: false,
+
+	// Custom types are used to define additional fields in package.json that contain versions that should be
+	// checked/synced. See https://github.com/JamieMason/syncpack/blob/master/README.md#customtypes for more details.
+	customTypes: {
+		enginesNpm: {
+			path: "engines.npm",
+			strategy: "version",
+		},
+		enginesNode: {
+			path: "engines.node",
+			strategy: "version",
+		},
+		packageManager: {
+			path: "packageManager",
+			strategy: "name@version",
+		},
+	},
+
+	/**
+	 * SemverGroups are used to ensure that groups of packages use the same semver range for dependencies.
+	 *
+	 * semverGroup rules are applied in order to package/dep combinations. First matching rule applies. When running
+	 * `syncpack lint-semver-ranges`, the output is grouped into numbered groups. However, the numbers of the groups are
+	 * the _inverse_ of their order in this array. That is, Semver Group 1 in the output will correspond to the _last_
+	 * rule in this array.
+	 */
+	semverGroups: [
+		// GROUP 8
+		// engines.node should always use >= range
+		{
+			dependencyTypes: ["enginesNode"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: ">=",
+		},
+
+		// GROUP 7
+		// engines.npm should always use ^ range
+		{
+			dependencyTypes: ["enginesNpm"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: "^",
+		},
+
+		// GROUP 6
+		// packageManager should always use exact version
+		{
+			dependencyTypes: ["packageManager"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: "",
+		},
+
+		// GROUP 5
+		// PropertyDDS packages' dependencies are ignored because they use a lot of exact deps.
+		{
+			dependencies: ["**"],
+			packages: ["@fluid-experimental/property-*"],
+			isIgnored: true,
+		},
+
+		// GROUP 4
+		// Dependencies declared in pnpm overrides should use caret.
+		{
+			dependencyTypes: ["pnpmOverrides"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: "^",
+		},
+
+		// GROUP 3
+		// These dependencies should always be on exact versions
+		{
+			dependencies: [
+				"@tiny-calc/*",
+				"@graphql-codegen/cli",
+				"@graphql-codegen/typescript",
+				"@types/chrome",
```

Comment:
I'm curious as to the motivation for pinning the @types packages here, but not their corresponding runtime packages. Don't we want them to "match" / get updated together?

Reply:
I tried to follow the conventions we have in place - which may have no reasoning. Updating these deps should be done separately from this PR, and probably individually because lots of things broke when I tested it.

All the deps updates in this PR narrow the dep range, with the exception of re-resizable, which went from an exact version to a `^` But everything else went from `^` to `~` or exact versions. That's one reason I feel safer about them.

Reply:
Any plans to follow up on this later? This definitely feels like something we should address. But since these are existing patterns, I have no worries about deferring it outside of this PR.